modules/auth/proc-adding-google-as-an-authentication-provider.adoc

```diff
@@ -0,0 +1,93 @@
+[id='proc-adding-google-as-an-authentication-provider_{context}']
+= Adding Google as an authentication provider in {product}
+
+The {product} `core-plugin-api` package includes a Google authentication provider that can authenticate users by using Google OAuth2.
+
+.Procedure
+. To allow {product} to authenticate with Google, create OAuth credentials in Google.
+
+
+.. Log in to the link:https://console.cloud.google.com/[Google Console].
+.. Select or create a new project from the dropdown menu on the top bar.
+.. Go to link:https://console.cloud.google.com/apis/credentials/consent[*APIs & Services > OAuth consent screen*],
+and click *Edit app* to configure the consent screen:
+**Authorized domains**::
+For local development:: You do not need to enter any Authorized domain.
+For a non-local deployment::: Enter your {product} domain.
+*Scopes*:: Add these scopes:
+`openid`::: Associate you with your personal info on Google.
+`auth/userinfo.email`::: See your primary Google Account email address.
+`auth/userinfo.profile`::: See your personal info, including any personal info you've made publicly available.
+*User type*:: Add yourself as a test user, if using the `External user` type.
+.. Go to to link:https://console.cloud.google.com/apis/credentials[*APIs & Services > Credentials*],
+click *Create Credentials*, choose *OAuth client ID*, and create a credential with following parameters:
+*Application type*:: Select `Web Application`.
+*Name*:: Enter your application name, such as `{product}`.
+*Authorized JavaScript origins*:: Add URI: `pass:[http://localhost:3000]`
+*Authorized redirect URIs*:: Click *Add URI* to enter the backend authentication URI set in {product}:
+For local development::: `pass:[http://localhost:7007/api/auth/google/handler/frame]`
+For a non-local deployment::: `pass:[http://{APP_FQDN}:{APP_BACKEND_PORT}/api/auth/google/handler/frame]`
```

Comment:
for non local deployment it will be only http://{rhdh_insteance_fqdn}/api/auth/google/handler/frame

btw, I've noticed that we are using different placeholder for RHDH the hostname of the RHDH instance that is getting configured. It would be nice if we could unify that

Reply:
we can add more reusable strings into https://github.com/redhat-developer/red-hat-developers-documentation-rhdh/blob/main/artifacts/attributes.adoc to make life easier.

note that if you're then going to use those variables in a code snippet you need `subs="attributes"` in your code block. See examples:

https://github.com/redhat-developer/red-hat-developers-documentation-rhdh/pull/99/files#diff-0f7e8a8fb3fdb835a98a8a937c0c87986260380b80b77a1f84f8ca529be430a2

Reply:
```suggestion
For a non-local deployment::: `pass:[http://{APP_FQDN}/api/auth/google/handler/frame]`
```